lib/src/utils/event_localizations.dart

```diff
@@ -114,89 +115,62 @@ abstract class EventLocalizations {
+          return i18n.userLeftTheChat(targetName);
+        case RoomMemberChangeType.kick:
+          userIsSender
               ? i18n.youKicked(targetName)
               : i18n.kicked(senderName, targetName);
-        } else if (oldMembership == 'join' &&
-            newMembership == 'leave' &&
-            event.stateKey == event.senderId) {
-          text = i18n.userLeftTheChat(targetName);
-        } else if (oldMembership == 'invite' && newMembership == 'ban') {
-          text = userIsSender
-              ? i18n.youBannedUser(targetName)
-              : i18n.bannedUser(senderName, targetName);
-        } else if (oldMembership == 'leave' && newMembership == 'ban') {
-          text = userIsSender
-              ? i18n.youBannedUser(targetName)
-              : i18n.bannedUser(senderName, targetName);
-        } else if (oldMembership == 'ban' && newMembership == 'leave') {
-          text = userIsSender
-              ? i18n.youUnbannedUser(targetName)
-              : i18n.unbannedUser(senderName, targetName);
-        } else if (newMembership == 'invite') {
-          text = userIsSender
+        case RoomMemberChangeType.invite:
+          return userIsSender
               ? i18n.youInvitedUser(targetName)
               : userIsTarget
                   ? i18n.youInvitedBy(senderName)
                   : i18n.invitedUser(senderName, targetName);
-        } else if (newMembership == 'join') {
-          text = userIsTarget
+        case RoomMemberChangeType.ban:
+          return userIsSender
+              ? i18n.youBannedUser(targetName)
+              : i18n.bannedUser(senderName, targetName);
+        case RoomMemberChangeType.unban:
+          return userIsSender
+              ? i18n.youUnbannedUser(targetName)
+              : i18n.unbannedUser(senderName, targetName);
+        case RoomMemberChangeType.knock:
+          return i18n.hasKnocked(targetName);
+        case RoomMemberChangeType.other:
+        default:
+          return userIsTarget
               ? i18n.youJoinedTheChat
               : i18n.joinedTheChat(targetName);
-        }
-      } else if (newMembership == 'join') {
-        final newAvatar = event.content.tryGet<String>('avatar_url') ?? '';
-        final oldAvatar = event.prevContent?.tryGet<String>('avatar_url') ?? '';
-
-        final newDisplayname =
-            event.content.tryGet<String>('displayname') ?? '';
-        final oldDisplayname =
-            event.prevContent?.tryGet<String>('displayname') ?? '';
-        final stateKey = event.stateKey;
-
-        // Has the user avatar changed?
-        if (newAvatar != oldAvatar) {
-          text = i18n.changedTheProfileAvatar(targetName);
-        }
-        // Has the user displayname changed?
-        else if (newDisplayname != oldDisplayname && stateKey != null) {
-          text = i18n.changedTheDisplaynameTo(oldDisplayname, newDisplayname);
-        }
       }
-      return text;
+      return userIsTarget
```

Comment:
Isn't that already in the default case above and would be done anyway?

Reply:
Fixed